File: 201415/J153.cpp
#include <cstdio>
#include <cstring>
char a[100001], b[100001];
int main() {
  int x, y;
  scanf("%s %s", a, b);
  x = strlen(a);
  y = strlen(b);
  int overtype = 0;
  int deleted = 0;
  for (int i = 0; i < y; i++) {
    if (b[i] == 'D') {
      deleted++;
    } else if (b[i] == 'I') {
      overtype = 1 - overtype;
    } else {
      printf("%c", b[i]);
      deleted += overtype;
    }
  }
  if (deleted < x) {
    printf("%s", a + deleted);
  }
  printf("\n");
  return 0;
}

File: 201415/J153.pas
var
  a, b: ansistring;
  i, y, overtype, deleted: longint;
begin
  readln(a);
  readln(b);
  y := length(b);
  overtype := 0;
  deleted := 0;
  for i := 1 to y do
  begin
    if (b[i] = 'D') then
      inc(deleted)
    else if (b[i] = 'I') then
      overtype := 1 - overtype
    else
    begin
      write(b[i]);
      deleted := deleted + overtype
    end
  end;
  delete(a, 1, deleted);
  writeln(a)
end.
